src/data/polynomial/eval.lean

```diff
@@ -447,6 +447,47 @@ lemma map_prod {ι : Type*} (g : ι → polynomial R) (s : finset ι) :
   (∏ i in s, g i).map f = ∏ i in s, (g i).map f :=
 eq.symm $ prod_hom _ _
 
+lemma map_sum {ι : Type*} (g : ι → polynomial R) (s : finset ι) :
+  (∑ i in s, g i).map f = ∑ i in s, (g i).map f :=
+eq.symm $ sum_hom _ _
+
+lemma map_support_is_subset (p : polynomial R) : (map f p).support ⊆ p.support :=
+begin
+  intros x,
+  simp only [mem_support_iff, classical.not_not] at *,
+  contrapose!,
+  change p.coeff x = 0 → (map f p).coeff x = 0,
+  rw coeff_map,
+  intro hx,
+  rw hx,
+  exact ring_hom.map_zero f,
+end
+
+lemma map_comp (p q : polynomial R) : map f (p.comp q) = (map f p).comp (map f q) :=
```

Comment:
I wrote a shorter proof of this statement.
```lean
lemma map_comp (p q : polynomial R) : map f (p.comp q) = (map f p).comp (map f q) :=
polynomial.induction_on p 
  (by simp)
  (by simp {contextual := tt}) 
  (by simp [pow_succ', ← mul_assoc] {contextual := tt})
```
Incidentally, I think the best proof would be to prove that homomorphisms out of `polynomial R` l ring are equal iff they're equal on `X` and equal on `C`, and then use the fact that `λ p, map f (p.comp q)` is a ring homomorphism. Both generated goals should be very easy after that. The library isn't set up like this right now however, and it would require bundling morphisms in the polynomial file, but it might make your life easier.

Reply:
I will change the proof of `map_comp` to your shorter proof. I don't really know how things like bundling work so I probably can't currently implement your other suggestion.

Reply:
For me, the proof you gave doesn't work. The third `simp` doesn't solve the goal.

Reply:
Okay, committed.